app/scripts/services/pods.js

```diff
@@ -40,6 +40,34 @@ angular.module("openshiftConsole")
         return debugPod;
       },
 
+      // Takes in a collection of pods and a collection of resources that
+      // control pods (replication controllers, replica sets, pet sets) and
+      // matches each pod to its owner.
+      //
+      // Returns: hash where the UID of the owner is the key and a hash of pods
+      //          is the value
+      groupByOwnerUID: function(pods, owners) {
+        var podsByOwnerUID = {};
+        // Look at each pod.
+        _.each(pods, function(pod) {
+          var found = false;
+          _.each(owners, function(owner) {
+            var selector = new LabelSelector(owner.spec.selector);
+            if (selector.matches(pod)) {
+              _.set(podsByOwnerUID, [owner.metadata.uid, pod.metadata.name], pod);
+              found = true;
+              return false;
```

Comment:
i'm wondering now whether we should be breaking out early like this, I know this isn't new logic, but in theory you can have more than one owner for a pod if you end up with two controllers fighting over the same pod.  Fighting controllers will be fixed eventually but not in 1.4.  Frankly multiple pod owners should probably be showing up as a warning.